fmc/test-fw/test-rt/src/main.rs

```diff
@@ -98,7 +118,12 @@ fn fmc_panic(_: &core::panic::PanicInfo) -> ! {
 fn report_error(code: u32) -> ! {
     cprintln!("RT Error: 0x{:08X}", code);
     report_fw_error_non_fatal(code);
-    loop {}
+    loop {
+        // Poorly-written SoC firmware might be stuck waiting for Caliptra
+        // to finish executing this pending mailbox transaction. Notify them
+        // that we've failed.
+        unsafe { Mailbox::abort_pending_soc_to_uc_transactions() };
```

Comment:
Hi, Kor. Is there a test case that triggers this failure mode, so we can compare the behavior with the fix and without the fix?

Reply:
Yeah, pretty much any of @mhatrevi's failure test cases exercise this behavior:

https://github.com/chipsalliance/caliptra-sw/blob/main/rom/dev/tests/test_image_validation.rs#L141-L148

The test passes as-is, but after this PR is merged you could get rid of the wait-for-fatal error logic and just check the error-code directly:

https://github.com/chipsalliance/caliptra-sw/blob/main/rom/dev/tests/test_image_validation.rs#L141-L148

could become

```
assert_eq!(
        Err(ModelError::MailboxCmdFailed),
        hw.upload_firmware(&image_bundle.to_bytes().unwrap())
    );
assert_eq!(hw.soc_ifc().cptra_fw_error_non_fatal().read(), 
           OWNER_PUB_KEY_DIGEST_MISMATCH);
```

Reply:
(In case it wasn't clear, removing this code will cause all of those failure tests to freeze as hw.upload_firmware() never gets a response).

Reply:
The error scenario  is clear.